### PR TITLE
Favorite makers

### PR DESCRIPTION
**Pull request by anhthang**
Add/remove a maker to favorite list and put it on top. So you can access it quickly

image

### CODE REVIEW

File: src/internal/favorite.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { localStorageLoad, localStorageSet } from './misc';
+
+const CONSTS = {
+  favorite: 'favorite',
```
Comment (zekth) [inline src/internal/favorite.js:4]: Shouldn't this be `favoriteMakers` ?

File: src/pages/index.js [hunk 2]
```diff
@@ -42,18 +45,47 @@ const IndexPage = () => {
     return img.find((x) => x.name === 'nologo').childImageSharp.fluid;
   };
 
+  const [favoriteMakers, setFavoriteMakers] = useState(getFavoriteMakers());
+
+  const sortedMakers = sortBy(data.allSitePage.nodes, (n) => !favoriteMakers.includes(n.context.maker.id));
```
Comment (zekth) [inline src/pages/index.js:50]: I wonder if this line that make the CI fails

**Review by zekth — CHANGES_REQUESTED** [review]

File: src/internal/favorite.js (hunk 1 above)
Comment (anhthang) [reply src/internal/favorite.js:4]: will change

File: src/pages/index.js (hunk 2 above)
Comment (anhthang) [reply src/pages/index.js:50]: do you mean this error in the CI?

image

Comment (zekth) [reply src/pages/index.js:50]: Yes

Comment (anhthang) [reply src/pages/index.js:50]: I have debugged, this sculpt cause that issue. Do you want to fix it in the database also?

image

File: gatsby-node.js [hunk 3]
```diff
@@ -47,7 +47,7 @@ exports.createPages = async ({ graphql, actions }) => {
       const f = element.colorways.find((x) => x.isCover === true);
       if (f) {
         element.previewImg = `https://cdn.keycap-archivist.com/keycaps/250/${f.id}.jpg`;
-      } else {
+      } else if (element.colorways[rng]) {
```
Comment (zekth) [inline gatsby-node.js:50]: Why would there be no colorway?

Comment (anhthang) [reply gatsby-node.js:50]: a custom import (gooey keys) returning no colorway.

**Review by zekth — APPROVED** [review]

**Pull request merged**